src/main/java/TaskManager.java

```diff
@@ -4,15 +4,42 @@
     protected static Task[] tasks = new Task[100];
     protected static int inputAmount = 0;
 
-    public void addTask(Task task){
-        tasks[inputAmount++] = task;
+    public static void addTask(String reply){
+        String choice = reply.split(" ")[0];
+        String newTask = reply.replace(choice+" ", "");
+        switch(choice.toUpperCase()) {
+            case "TODO":
+                ToDo todo = new ToDo(newTask);
+                tasks[inputAmount++] = todo;
+                break;
+            case "DEADLINE":
+                int startIndexforBy = newTask.indexOf("/by");
+                String by = newTask.substring(startIndexforBy + 4);
+                String newDeadline = newTask.substring(0, startIndexforBy - 1);
+                Deadline deadline = new Deadline(newDeadline, by);
+                tasks[inputAmount++] = deadline;
+                break;
+            case "EVENT":
+                int startIndexforAt = newTask.indexOf("/at");
+                String at = newTask.substring(startIndexforAt + 4);
+                String newEvent = newTask.substring(0, startIndexforAt - 1);
+                Event event = new Event(newEvent, at);
+                tasks[inputAmount++] = event;
+                break;
+        }
+        System.out.println("____________________________________________________________");
+        System.out.println("Got it. I've added this task: ");
+        System.out.println(tasks[inputAmount-1]);
+        System.out.println("Now you have " + inputAmount + " tasks in the list.");
+        System.out.println("____________________________________________________________");
```

Comment:
Perhaps method extraction can be applied here and to other content in this method to make it even neater

Reply:
Thank you for your suggestion! I will refactor my code of this part later ^_^